File: VB/Demo.Module/Module.vb
Imports System
Imports System.Collections.Generic

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports Demo.Module.BusinessObjects

Namespace Demo.Module
    Public NotInheritable Partial Class DemoModule
        Inherits ModuleBase

        Public Sub New()
            InitializeComponent()
        End Sub
        Public Overrides Function GetModuleUpdaters(ByVal objectSpace As IObjectSpace, ByVal versionFromDB As Version) As IEnumerable(Of ModuleUpdater)
            Dim updater As ModuleUpdater = New DatabaseUpdate.Updater(objectSpace, versionFromDB)
            Return New ModuleUpdater() { updater }
        End Function
        Public Overrides Sub Setup(ByVal application As XafApplication)
            MyBase.Setup(application)
            XafTypesInfo.Instance.RegisterEntity("Document", GetType(IDocument), GetType(GenerateUserFriendlyId.Module.BusinessObjects.UserFriendlyIdPersistentObject))
        End Sub
    End Class
End Namespace

File: CS/XPO/Demo.Module/Module.cs
using System;
using System.Collections.Generic;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using Demo.Module.BusinessObjects;

namespace Demo.Module {
    public sealed partial class DemoModule : ModuleBase {
        public DemoModule() {
            InitializeComponent();
        }
        public override IEnumerable<ModuleUpdater> GetModuleUpdaters(IObjectSpace objectSpace, Version versionFromDB) {
            ModuleUpdater updater = new DatabaseUpdate.Updater(objectSpace, versionFromDB);
            return new ModuleUpdater[] { updater };
        }
        public override void Setup(XafApplication application) {
            base.Setup(application);
            XafTypesInfo.Instance.RegisterEntity("Document", typeof(IDocument), typeof(GenerateUserFriendlyId.Module.BusinessObjects.UserFriendlyIdPersistentObject));
        }
    }
}
